server/api.go

```diff
@@ -665,6 +676,283 @@ func (a *API) getConnectedUsersFile(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (a *API) resetMocks(w http.ResponseWriter, r *http.Request) {
+	clientMock = nil
+	newMock := getClientMock(a.p)
+	a.p.msteamsAppClient = newMock
+	w.WriteHeader(http.StatusOK)
+}
+
+func (a *API) addMockCall(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	methodName := params["method"]
+
+	mockClient := a.p.clientBuilderWithToken("", "", "", "", nil, nil).(*mocks.Client)
+	method, found := reflect.TypeOf(mockClient).MethodByName(methodName)
+	if !found {
+		a.p.API.LogError("Unable to mock the method, method not found", "MethodName", methodName)
+		http.Error(w, "method not found", http.StatusNotFound)
+		return
+	}
+
+	paramsCount := method.Type.NumIn()
+	parameters := []interface{}{}
+	for x := 0; x < paramsCount; x++ {
+		parameters = append(parameters, mock.Anything)
+	}
+
+	var mockCall testmodels.MockCallReturns
+	err := json.NewDecoder(r.Body).Decode(&mockCall)
+	if err != nil {
+		http.Error(w, "unable to mock the method", http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+	var returnErr error
+	if mockCall.Err != "" {
+		returnErr = errors.New(mockCall.Err)
+	}
+
+	a.p.API.LogDebug("mocking method", "method", method, "returns", mockCall.Returns, "returnType", mockCall.ReturnType, "returnErr", returnErr)
+
+	var output any
+
+	returns := method.Type.NumOut()
+	switch returns {
```

Comment:
There is a lot of duplication here. Can we clean this up?

Reply:
Done, will be available whenever I push myt local changes.